odtools/odcomparepointcloud/src/ComparePointCloudModule.cpp

```diff
@@ -0,0 +1,373 @@
+                    }
+                }
+                 
+                float max_xError=m_xError[0];
+                float sum_xError=0.0;
+                for(float i : m_xError){
+                    sum_xError+=i;
+                    if(i>max_xError){
+                        max_xError=i;
+                    }
+                }
+                float avg_xError=sum_xError/m_xError.size();
+               
+                float max_yError=m_yError[0];
+                float sum_yError=0.0;
+                for(float i : m_yError){
+                    sum_yError+=i;
+                    if(i>max_yError){
+                        max_yError=i;
+                    }
+                }
+                float avg_yError=sum_yError/m_yError.size();
+                
+                float max_zError=m_zError[0];
+                float sum_zError=0.0;
+                for(float i : m_zError){
+                    sum_zError+=i;
+                    if(i>max_zError){
+                        max_zError=i;
+                    }
+                }
+                float avg_zError=sum_zError/m_zError.size();
+                if(m_xError.size()==m_xSpc.size() && m_yError.size()==m_ySpc.size() && 
+                m_zError.size()==m_zSpc.size()){
+                    m_outputData<<max_xError<<","<<avg_xError<<","<<max_yError<<","<<avg_yError<<","
+                    <<max_zError<<","<<avg_zError<<endl;  
+                }
+                else{
+                   cout<<"Abnormal frame:"<<m_frameNumber<<endl;
+                }
+                clearVectors();   
+            }
+        }
+        else{
+            /*if(player->hasMoreData()){
+                //CPC has one more frame than SPC. Discard the first frame of CPC
+                c = player->getNextContainerToBeSent();
+            }*/
+            while(player->hasMoreData() && m_currentFrame<m_chosenFrame){
+                c = player->getNextContainerToBeSent();
+                c = player->getNextContainerToBeSent();
+                m_currentFrame++;
+            }
+            
+            if(m_currentFrame<m_chosenFrame){
+                cerr<<"The chosen frame number is too large for the recording. Choose a smaller frame."<<endl;
+            }
+            else{
+                //Compare x, y, z of the chosen frame between CPC and SPC
+                spcFrame = player->getNextContainerToBeSent();
+                if(spcFrame.getDataType() == odcore::data::SharedPointCloud::ID()){
+                    readSPC(spcFrame);
+                }  
+                cout<<"Number of points of Frame "<<m_chosenFrame<<" of SPC:"<<m_xSpc.size()<<endl;
+                
+                cpcFrame = player->getNextContainerToBeSent();
+                if(cpcFrame.getDataType() == odcore::data::CompactPointCloud::ID()){
+                    readCPC(cpcFrame);
+                }  
+                cout<<"Number of points of Frame "<<m_chosenFrame<<" of CPC:"<<m_xCpc.size()<<endl;
```

Comment:
You should ensure to compare the time stamps for the sampleTimeStamp as well (cf. my comment in OpenDLV.core/proxy-Velodyne)